modules/nextflow/src/main/groovy/nextflow/script/ProcessConfig.groovy

```diff
@@ -65,6 +65,7 @@ class ProcessConfig implements Map<String,Object>, Cloneable {
             'machineType',
             'queue',
             'label',
+            'maxAwait',
```

Comment:
Think a better name would be `maxSubmitTimeout`

Reply:
Also maybe should be an executor config option instead of process directive, like all the other timeout / interval settings

Reply:
Feel `maxAwait` is more concise. 

> Also maybe should be an executor config option instead of a process directive

Process directives are more powerful because they allow finer control. Usually, it's a defined a executor level, when there isn't a direct process mapping or there's limitation having the setting a process level.